test/rules/non_constant_identifier_names_test.dart

```diff
@@ -19,6 +19,16 @@ class NonConstantIdentifierNamesPatternsTest extends LintRuleTest {
   @override
   String get lintRule => 'non_constant_identifier_names';
 
+  test_identifierPattern() async {
+    await assertNoDiagnostics(r'''
+f() {
+  var a = 0;
```

Comment:
Hmm, this is not new syntax, so I wouldn't expect anything to be surprising here. But a variable declaration pattern might not get caught: `var [_, a] = [1, 2];`

Reply:
> Hmm, this is not new syntax

I believe this is a "wildcard pattern" 

https://github.com/dart-lang/language/blob/master/accepted/future-releases/0546-patterns/feature-specification.md#wildcards

And this is a different case 

> var [_, snake_case] = [1, 2];

(where `_` is being declared as a local variable).

The question is whether all-underscores should be ok in this case.

I'll look into this today.

Reply:
As for the wildcard, I was just avoiding an `unused_local_variable` issue, we could remove that; `snake_case` is the important part.